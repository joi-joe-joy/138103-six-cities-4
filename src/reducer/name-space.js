export const NameSpace = {
  DATA: `DATA`,
  USER: `USER`,
  PLACE: `PLACE`
};
