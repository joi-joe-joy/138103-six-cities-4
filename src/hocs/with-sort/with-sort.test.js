import React from 'react';
import renderer from "react-test-renderer";
import withSort from "./with-sort";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {NameSpace} from "../../reducer/name-space.js";
import pt from "prop-types";

const MockComponent = (props) => {
  const {children} = props;

  return <div>{children}</div>;
};

MockComponent.propTypes = {
  children: pt.oneOfType([
    pt.arrayOf(pt.node),
    pt.node
  ]).isRequired
};

const mockStore = configureStore([]);
const MockComponentWrap = withSort(MockComponent);

it(`should render withSort correctly`, () => {
  const store = mockStore({
    [NameSpace.PLACE]: {
      sorting: `popular`
    }
  });

  const tree = renderer.create(
      <Provider store={store}>
        <MockComponentWrap
          changeSorting={()=>{}}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
