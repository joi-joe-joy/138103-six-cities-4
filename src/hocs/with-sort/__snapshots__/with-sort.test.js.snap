// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render withSort correctly 1`] = `
<div>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      height="4"
      style={
        Object {
          "marginBottom": "1px",
          "marginLeft": "5px",
        }
      }
      width="7"
    />
  </span>
  <ul
    className="places__options places__options--custom"
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</div>
`;
